ams/optimizer/src/main/java/com/netease/arctic/optimizer/flink/FlinkOptimizer.java

```diff
@@ -36,4 +44,25 @@ public static void main(String[] args) throws CmdLineException {
       LOG.error("Execute flink optimizer failed", e);
     }
   }
+
+  private static void calcOptimizerMemory(OptimizerConfig config, StreamExecutionEnvironment env) {
+    ReadableConfig configuration = env.getConfiguration();
+    MemorySize jobMemorySize = configuration.get(JobManagerOptions.TOTAL_PROCESS_MEMORY);
+    MemorySize taskMemorySize = configuration.get(TaskManagerOptions.TOTAL_PROCESS_MEMORY);
+    if (jobMemorySize == null || taskMemorySize == null) {
+      // Running locally will not get the configuration.
```

Comment:
Is `Running locally` means deploying the flink job in the standalone cluster?

Besides, how can we calculate the memory usage in this situation if we have deprecated the `-m` configuration?

Reply:
> Is `Running locally` means deploying the flink job in the standalone cluster?

`Running locally` means running the optimizer in the IDEA. This method is supported in all deployment modes.

> Besides, how can we calculate the memory usage in this situation if we have deprecated the `-m` configuration?

In this situation, there is indeed no coverage. We can also obtain the usage of jvm in the same way as the local optimizer. Do you think it is necessary?

Reply:
Ok, I don't think we will run the flink optimizer in the IDEA.

If we do, we still can use an argument to specify the memory usage.